File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/publicapi/profile/dto/PublicProfileDto.kt
package ch.streckeisen.mycv.backend.publicapi.profile.dto

import ch.streckeisen.mycv.backend.cv.project.PublicProjectDto

data class PublicProfileDto(
    val firstName: String,
    val lastName: String,
    val jobTitle: String,
    val bio: String?,
    val email: String?,
    val phone: String?,
    val address: PublicAddressDto?,
    val profilePicture: String,
    val workExperiences: List<PublicWorkExperienceDto>,
    val skills: List<PublicSkillDto>,
    val education: List<PublicEducationDto>,
    val projects: List<PublicProjectDto>,
    val theme: PublicProfileThemeDto?
)

File: frontend/src/types/profile/PublicProfileDto.ts
import { PublicWorkExperienceDto } from '@/types/profile/workExperience/PublicWorkExperienceDto.ts'
import { PublicSkillDto } from '@/types/profile/skill/PublicSkillDto.ts'
import { PublicEducationDto } from '@/types/profile/education/PublicEducationDto.ts'
import { PublicProjectDto } from '@/types/profile/project/PublicProjectDto.ts'
import { PublicProfileThemeDto } from '@/types/profile/theme/PublicProfileThemeDto.ts'


export interface PublicProfileDto {
  firstName: string
  lastName: string
  jobTitle: string
  bio: string
  email?: string
  phone?: string
  address?: PublicAddressDto
  profilePicture: string
  workExperiences: PublicWorkExperienceDto[]
  skills: PublicSkillDto[]
  education: PublicEducationDto[]
  projects: PublicProjectDto[]
  theme?: PublicProfileThemeDto
  language: string,
}

export interface PublicAddressDto {
  street: string
  houseNumber?: string
  postcode: string
  city: string
  country: string
}
